packages/react-components/react-drawer/docs/Spec.md

```diff
@@ -2,62 +2,258 @@
 
 ## Background
 
-_Description and use cases of this component_
+A Drawer's main function is to contain supplementary content and are used for complex creation, edit, or management experiences. For example, viewing details about an item in a list or editing settings. It is also useful to host navigational elements that can be visible at all times.
+
+The Drawer don't have any content built in, but has mechanisms to display it in a structured way.
 
 ## Prior Art
 
-_Include background research done for this component_
+- [Convergence Epic](https://github.com/microsoft/fluentui/issues/26420)
+
+### Fabric (v8)
+
+The [Panel component](https://developer.microsoft.com/en-us/fluentui#/controls/web/panel) on v8's version is very similar to what Drawer component is structured. It has an opinionated way to provide content, like header text and a close button.
+
+```tsx
+<Panel headerText="Sample panel" isOpen={isOpen} onDismiss={dismissPanel} closeButtonAriaLabel="Close">
+  <p>Content goes here.</p>
+</Panel>
+```
 
-- _Link to Open UI research_
-- _Link to comparison of v7 and v0_
-- _Link to GitHub epic issue for the converged component_
+### Northstar (v0)
+
+There is not prior implementation for this component in v0
+
+### 3rd party Design Systems
+
+- [Chakra UI](https://chakra-ui.com/docs/components/drawer)
+- Material Design
+  - [Official implementation](https://github.com/material-components/material-components-web-react/tree/master)
+  - [Material UI](https://mui.com/material-ui/react-drawer/)
+- [Radix UI](https://design-system.modulz-deploys.com/#sheet)
+  - [Sheet component](https://github.com/radix-ui/design-system/blob/master/components/Sheet.tsx)
+- [React Suite](https://rsuitejs.com/components/drawer/)
+- [Ant Design](https://ant.design/components/drawer)
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Drawer open={open} onOpenChange={(_, { open }) => setOpen(open)}>
+  <p>Content goes here.</p>
+</Drawer>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+### Type
+
+`overlay`:
+Opens on top of everything like a dialog and blocks all the page content. Can be dismissed.
+![Example of Drawer as overlay](assets/overlay.png)
+
+`inline`:
+Push the siblings content when open and it is non-blocking. Can be hidden to bring focus to the main content of the page.
+![Example of Drawer as inline content](assets/inline.png)
+
+### Size
+
+- `small` (default): 320px
+- `medium`: 592px
+- `large`: 940px
+- `full`: 100vw
+- `custom`: Can be freely customized by providing a number. It cannot be extended beyond screen limits. If the size is larger than the screen size, it'll act as a `full` Drawer.
+
+### Modal
+
+By default, the `overlay` acts as a modal, rendering an overlay scrim behind the drawer surface. This can be toggled off. \*
```

Comment:
How toggle is working? Will it be there only `close` button? WIll it be possible to close Drawer by clicking outside its area?